tests/test_frompyobject.rs

```diff
@@ -481,3 +481,43 @@ fn test_from_py_with() {
         assert_eq!(zap.some_object_length, 3usize);
     });
 }
+
+#[derive(Debug, FromPyObject)]
+pub struct ZapTuple(
+    #[pyo3(item)] String,
```

Comment:
You are right, the `#[pyo3(item)]` annotation didn't belonged here. During implementation I've been thinking about it and it might indeed be reasonable to throw an error `#[pyo3(item)]` or `#[pyo3(attribute)]` are used on tuple struct elements. That might be a nice follow-up issue.